datafusion/optimizer/src/filter_push_down.rs

```diff
@@ -248,6 +249,145 @@ fn get_pushable_join_predicates<'a>(
         .unzip()
 }
 
+// examine OR clause to see if any useful clauses can be extracted and push down.
+// extract at least one qual of each sub clauses of OR clause, then form the quals
+// to new OR clause as predicate.
+//
```

Comment:
I think we need to explain the conditions under which a qual can be extracted as it may not be obvious to someone when they initially look at this. 

```suggestion
// to new OR clause as predicate.
//
// A qual is extracted if it it contains (only) common set of column references with the other quals.
```

I am not sure that is correct

Reply:
I would also like to see the return type documented here (as in what does the `(Vec<Expr>, Vec<HashSet<Column>>)` represent? I think it is the extracted quals and their column references but I am not sure